web_ui/ui.go

```diff
@@ -137,12 +146,20 @@ func configureWebResource(engine *gin.Engine) error {
 
 		// If just one server is enabled, redirect to that server
 		if len(config.GetEnabledServerString(true)) == 1 && path == "/index.html" {
-			ctx.Redirect(http.StatusFound, "/view/"+config.GetEnabledServerString(true)[0]+"/index.html")
+			ctx.Redirect(http.StatusFound, "/view/"+config.GetEnabledServerString(true)[0]+"/")
 			return
 		}
 
 		filePath := "frontend/out" + path
 		file, _ := webAssets.ReadFile(filePath)
+
+		// If the file is not found, return 404
+		if file == nil {
+			ctx.AbortWithStatus(http.StatusNotFound)
```

Comment:
I think next.js generated a nice `404` page by default which we can use here: `/out/404/index.html`